doc/vlan/SONiC_VLAN_trunk_all_except_none_HLD.md

```diff
@@ -0,0 +1,701 @@
+!
+interface Ethernet4
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-4094
+sonic#
+```
+
+#### 3.3.2.3 IS-CLI Compliance
+All CLI commands mentioned above follows IS-CLI Compliance.
+
+### 3.3.3 REST API Support
+N/A
+
+# 4 Error Handling
+N/A
+
+# 5 Unit Test and automation
+The following test cases will be executed using KLISH CLI on Physical and Portchannel interfaces.
+
+#### 5.1 Test cases for “all”, “except”, “none” and *vlan-list* options: 
+##### 5.1.1 Test: switchport trunk allowed vlan all
+* Verify config using 'show run-configuration' output. 
+<br>*Sample output:*
+```
+sonic(conf-if-EthernetX)#switchport trunk allowed vlan all
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 10
+ switchport trunk allowed Vlan 1-4094
+```
+* Performance test by measuring the time taken for configuration.
+
+##### 5.1.2 Test: switchport trunk allowed vlan "except" *vlan-list*
+* Verify config using 'show run-configuration' output. 
+<br>*Sample output:*
+```
+sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 1000-4000
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-999,4001-4094
+```
+* Performance test by measuring the time taken for configuration.
+
+##### 5.1.3 Test: switchport trunk allowed vlan "none"
+* Verify config using 'show run-configuration' output. 
+<br>*Sample output:*
+```
+sonic(conf-if-EthernetX)#switchport trunk allowed vlan none
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+```
+* Performance test by measuring the time taken for configuration.
+
+##### 5.1.4 Test: switchport trunk allowed vlan *vlan-list* 
+* Verify config using 'show run-configuration' output. 
+<br>*Sample output:*
+```
+sonic(conf-if-EthernetX)#switchport trunk allowed vlan 100,1000-2000
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport access Vlan 10
+ switchport trunk allowed Vlan 100,1000-2000
+```
+* Performance test by measuring the time taken for configuration.
+
+#### 5.2 Test scenarios for “all”, “except”, “none” and *vlan-list* in combination with existing options: 
+
+**Scenario 1:**
+<br> Vlan 5 exists. 
+<br> Step 1: sonic(conf-if-EthernetX)#switchport access Vlan 5
+<br> Step 2: sonic(conf-if-EthernetX)#switchport trunk allowed vlan except 5
+<br> Result: Port remains untagged in Vlan5, port is tagged in vlan 1-4,6-4094 
+<br> Step 3: sonic(conf-if-EthernetX)#no switchport access vlan 
+<br> Result: Port tagged in vlan 1-4,6-4094 
+<br> Verify: 'show run-configuration' output.  
+Sample show run-config output:
+```
+sonic# show running-configuration interface Ethernet X
+!
+interface EthernetX
+ mtu 9100
+ speed 40000
+ shutdown
+ switchport trunk allowed Vlan 1-4,6-4094
```

Comment:
should include the line "switchport access Vlan 5

Reply:
"no switchport access vlan"  is also given(last step 3)